### PR TITLE
data source `azurerm_client_config` `azurerm_public_maintenance_configurations` `azurerm_public_ips` - Use static id instead of timestamp

### PR DESCRIPTION
**Pull request by myc2h6o**
Similar to #17766 and [shared_image_versions_data_source](https://github.com/hashicorp/terraform-provider-azurerm/blob/24d67dab775787d4d0418bec7b3101854fdd7dcf/internal/services/compute/shared_image_versions_data_source.go#L127), `time.Now()` changes on each apply, so if we reference the data source like `output "v" { value = data.azurerm_... }`, it will output a diff on each apply. Using the static id to fix this diff.

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result
image
image
image

File: internal/services/authorization/client_config_data_source.go [hunk 1]
```diff
@@ -45,7 +45,7 @@ func dataSourceArmClientConfigRead(d *pluginsdk.ResourceData, meta interface{})
 	_, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	d.SetId(time.Now().UTC().String())
+	d.SetId("armClientConfig")
```
Comment (katbyte) [inline internal/services/authorization/client_config_data_source.go:48]: can we make this a meaningful value? like
```suggestion
	d.SetId("clientConfig/$subscriptionid/$clientid/$objectid")
```

File: internal/services/network/public_ips_data_source.go [hunk 2]
```diff
@@ -125,7 +128,7 @@ func dataSourcePublicIPsRead(d *pluginsdk.ResourceData, meta interface{}) error
 		filteredIPAddresses = append(filteredIPAddresses, element)
 	}
 
-	d.SetId(time.Now().UTC().String())
+	d.SetId(fmt.Sprintf("publicIPs/resourceGroup:%s/namePrefix:%s/attachmentStatus:%s/allocationType:%s", resourceGroup, prefix, attachmentStatus, allocationType))
```
Comment (katbyte) [inline internal/services/network/public_ips_data_source.go:131]:
```suggestion
	d.SetId(fmt.Sprintf("network_publicIPs/resourceGroup/%s;namePrefix=%s;attachmentStatus=%s;allocationType=%s", resourceGroup, prefix, attachmentStatus, allocationType))
```

File: internal/services/maintenance/public_maintenance_configurations_data_source.go [hunk 3]
```diff
@@ -173,7 +174,7 @@ func dataSourcePublicMaintenanceConfigurationsRead(d *pluginsdk.ResourceData, me
 		return fmt.Errorf("setting `configs`: %+v", err)
 	}
 
-	d.SetId(time.Now().UTC().String())
+	d.SetId(fmt.Sprintf("publicMaintenanceConfigurations/location:%s/scope:%s/recurEvery:%s", locationFilter, scopeFilter, recurEveryFilterRaw))
```
Comment (katbyte) [inline internal/services/maintenance/public_maintenance_configurations_data_source.go:177]:
```suggestion
	d.SetId(fmt.Sprintf("publicMaintenanceConfigurations;location=%s;scope=%s;recurEvery=%s", locationFilter, scopeFilter, recurEveryFilterRaw))
```

**Review by katbyte — CHANGES_REQUESTED** [review]

**Comment by myc2h6o** [comment]
Hi @katbyte thanks for reviewing the pr! I've updated it id format according to the comment, please take a look.
The current id example is:
```
clientConfig/clientId=***;objectId=***;subscriptionId=***;tenantId=***
networkPublicIPs/resourceGroup/testrg/namePrefix=ipprefix;attachmentStatus=Unattached;allocationType=Static
publicMaintenanceConfigurations/location=eastus;scope=SQLManagedInstance;recurEvery=Friday-Sunday
```

File: internal/services/network/public_ips_data_source.go (hunk 2 above)
Comment (tombuildsstuff) [reply internal/services/network/public_ips_data_source.go]: this should be using a ARM Style Resource ID (or [base64] encoding this value prior to setting it into the state)

File: internal/services/maintenance/public_maintenance_configurations_data_source.go (hunk 3 above)
Comment (tombuildsstuff) [reply internal/services/maintenance/public_maintenance_configurations_data_source.go]: this should be using a ARM Style Resource ID (or [base64] encoding this value prior to setting it into the state)

**Comment by myc2h6o** [comment]
Hi @tombuildsstuff thanks for reviewing! I've updated the code to encode the id into base64 string, please take a look.

**Review by katbyte — APPROVED** [review]
LGTM 🧞

**Pull request merged**